var/spack/repos/builtin/packages/py-cython/package.py

```diff
@@ -8,23 +8,39 @@
 
 class PyCython(PythonPackage):
     """The Cython compiler for writing C extensions for the Python language."""
+
     homepage = "https://pypi.python.org/pypi/cython"
-    url      = "https://pypi.io/packages/source/c/cython/Cython-0.25.2.tar.gz"
+    url      = "https://pypi.io/packages/source/c/cython/Cython-0.29.7.tar.gz"
+
+    import_modules = [
+        'cython', 'Cython', 'Cython.Build', 'Cython.Compiler',
+        'Cython.Runtime', 'Cython.Distutils', 'Cython.Debugger',
+        'Cython.Debugger.Tests', 'Cython.Plex', 'Cython.Tests',
+        'Cython.Build.Tests', 'Cython.Compiler.Tests', 'Cython.Utility',
+        'Cython.Tempita', 'pyximport',
+    ]
 
+    version('0.29.7', sha256='55d081162191b7c11c7bfcb7c68e913827dfd5de6ecdbab1b99dab190586c1e8')
     version('0.29.5', sha256='9d5290d749099a8e446422adfb0aa2142c711284800fb1eb70f595101e32cbf1')
-    version('0.29', sha256='94916d1ede67682638d3cc0feb10648ff14dc51fb7a7f147f4fedce78eaaea97')
+    version('0.29',   sha256='94916d1ede67682638d3cc0feb10648ff14dc51fb7a7f147f4fedce78eaaea97')
     version('0.28.6', '3c3fb47806a4476f8e9429943439cc60')
     version('0.28.3', '586f0eb70ba1fcc34334e9e10c5e68c0')
     version('0.28.1', 'c549effadb52d90bdcb1affc1e5dbb97')
     version('0.25.2', '642c81285e1bb833b14ab3f439964086')
     version('0.23.5', '66b62989a67c55af016c916da36e7514')
     version('0.23.4', '157df1f69bcec6b56fd97e0f2e057f6e')
-
-    # These versions contain illegal Python3 code...
-    version('0.22', '1ae25add4ef7b63ee9b4af697300d6b6')
+    version('0.22',   '1ae25add4ef7b63ee9b4af697300d6b6')
     version('0.21.2', 'd21adb870c75680dc857cd05d41046a4')
 
+    depends_on('python@:2', type=('build', 'run'), when='@:0.22')
```

Comment:
Is that right?  Python 2.0?  Or is it supposed to be `python@:2.8`?

Reply:
`:2` corresponds to up to and including Python 2. Just like `:2.6` means up to and including Python 2.6. So this will include all versions of Python 2.

Reply:
Well that's certainly an oddity of versioning I didn't expect.  I assumed `python@:2` was functionally equivalent to `python@:2.0.0`, but you're saying that it's not in instead functionally equivalent to `python@:2.X.Y`.  That's neat, and super helpful (undocumented though I believe).